docs/test_specification.yaml

```diff
@@ -0,0 +1,218 @@
+definitions: 
+  hello1: 
+    - A signed hello_word image
+  hello2:
+    - An upgrade image, signed and marked for upgrade 
+  good_key:
+    - A key generated with a cryptosystem set in the bootloader with a correct value
+  bad_key:
+    - A key generated with a cryptosystem set in the bootloader with an incorrect value
+  wrong_key:   
```

Comment:
I'm starting thinking that `wrong_key` is wrong name for such a key. No much differ from `bad_key`. Maybe `incompatible_key`.

If decided to switch term, then Terminology should be aligned for testcases name as well.

@d3zd3z Any thought on terminology.

Reply:
I wasn't quite sure what to call it. There are a few things that should be tested here. An obvious one (that I think is missed) is where the hash/signature is generated from the wrong image. This will fail before the signature is even verified. Then there are several things that can be wrong with the signature itself. It could be things like this where the wrong key is just used. It ends up just showing up as a bad signature. It is also possible to do things like modify a bit in the signature, which will cause the signature to fail for a possibly different reason.

I'm not sure I understand why "wrong key" is any better than "bad key". There is also the possibility of testing what happens if we have the wrong kind of key (ECDSA signature but build configured for RSA).